### PR TITLE
Event and job fields add in Volunteer search form

### LINKED ISSUE #773: Add event and job field in Volunteer Search
## Description
As an administrator,
I need the functionality to search volunteers from event and job names as well.

## Acceptance Criteria
- [ ]  Volunteers can be searched using event or job name.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
1.5 hours

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
# Description
Add event and job in search volunteer form

Fixes #773 

# Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
image

# Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: vms/volunteer/views.py [hunk 1]
```diff
@@ -1,256 +1,261 @@
...
+from shift.services import get_volunteer_report, calculate_total_report_hours
+from volunteer.forms import ReportForm, SearchVolunteerForm, VolunteerForm
+from volunteer.models import Volunteer
+from volunteer.services import (delete_volunteer_resume, search_volunteers,
+                                get_volunteer_resume_file_url, has_resume_file,
+                                get_volunteer_by_id)
+from volunteer.validation import validate_file
+from volunteer.utils import vol_id_check
+from vms.utils import check_correct_volunteer
+import ipdb
```
[the remark this answers is not in the document]
Comment (kamsuri) [reply vms/volunteer/views.py]: Where have you used this?

File: vms/pom/locators/volunteerSearchPageLocators.py [hunk 2]
```diff
@@ -6,6 +6,8 @@ class VolunteerSearchPageLocators(object):
     STATE_FIELD = ".form-control[name='state']"
     COUNTRY_FIELD = ".form-control[name='country']"
     ORG_FIELD = ".form-control[name='organization']"
+    EVENT_FIELD = ".form-control[name='event']"
```
[the remark this answers is not in the document]
Comment (kamsuri) [reply vms/pom/locators/volunteerSearchPageLocators.py]: Fix this.

**Comment by abhishekarya286** [comment]
Please fix the conflicts.

**Comment by nida** [comment]
@anjali-dhanuka This is merge code but the corresponding PRs are not approved ? correct me ?

**Comment by abhishekarya286** [comment]
No @nida, the corresponding PRs have been approved.

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 @nida @kamsuri Done with merge conflicts, please review

File: vms/volunteer/views.py [hunk 3]
```diff
@@ -222,35 +222,41 @@ def post(self, request, *args, **kwargs):
...
                     'organizations_list': organizations_list,
+                    'events_list': events_list,
+                    'jobs_list': jobs_list,
                     'form': form,
                     'has_searched': True,
                     'search_result_list': search_result_list
                 })
     else:
-        organizations_list = get_organizations_ordered_by_name()
         form = SearchVolunteerForm()
```
[the remark this answers is not in the document]
Comment (anjali-dhanuka) [reply vms/volunteer/views.py]: By default, as you open the page it just shows the search form. This is already existing, it has nothing to do with this PR.

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 @nida @Monal5031 @kamsuri Please review

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 # local Django
 from pom.pages.authenticationPage import AuthenticationPage
 from pom.pages.volunteerSearchPage import VolunteerSearchPage
-from shift.utils import (create_admin, create_volunteer_with_details, create_organization_with_details)
+from shift.utils import (create_admin, create_volunteer_with_details,  create_organization_with_details, register_volunteer_for_shift_utility, register_event_utility, register_job_utility, register_shift_utility)
```
Comment (Monal5031) [inline vms/volunteer/tests/test_searchVolunteer.py]: two spaces after 1st comma?

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 5]
```diff
@@ -24,6 +24,8 @@ class SearchVolunteer(LiveServerTestCase):
     - State
     - Country
     - Organization
+    - Event
+    - Job
     Class contains 7 tests to check each parameter separately and also to check
```
Comment (Monal5031) [inline vms/volunteer/tests/test_searchVolunteer.py]: 2 more now?

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 6]
```diff
@@ -478,6 +480,107 @@ def test_volunteer_valid_organization_field(self):
         self.assertTrue(expected_result_one in result)
         self.assertTrue(expected_result_two in result)
 
+    def test_volunteer_event_field(self):
```
Comment (Monal5031) [inline vms/volunteer/tests/test_searchVolunteer.py]: Docstring

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 7]
```diff
@@ -478,6 +480,107 @@ def test_volunteer_valid_organization_field(self):
...
+        search_page.navigate_to_volunteer_search_page()
+        search_page.search_event_field("event")
+        search_page.submit_form()
+        search_results = search_page.get_search_results()
+        result = search_page.get_results_list(search_results)
+        self.assertEqual(len(result), 2)
+        self.assertTrue(expected_result_one in result)
+        self.assertTrue(expected_result_two in result)
+
+    def test_volunteer_job_field(self):
```
Comment (Monal5031) [inline vms/volunteer/tests/test_searchVolunteer.py]: docstring

File: vms/volunteer/tests/test_services.py [hunk 8]
```diff
@@ -141,16 +141,20 @@ def test_search_volunteers(self):
...
         self.v3.save()
 
+        event=register_event_utility()
+        job=register_job_utility()
+        shift=register_shift_utility()
+        register_volunteer_for_shift_utility(shift, self.v1)
         # if no search parameters are given,
         # it returns all volunteers
-        search_list = search_volunteers("", "", "", "", "", "")
+        search_list = search_volunteers("", "", "", "", "", "","","")
```
Comment (Monal5031) [inline vms/volunteer/tests/test_services.py]: space after commas.

**Comment by abhishekarya286** [comment]
image

For some reason, it returned the following volunteer twice after searching. I used `event` in the volunteer search form.

File: vms/volunteer/services.py [hunk 9]
```diff
@@ -140,5 +139,9 @@ def search_volunteers(first_name, last_name, city, state, country,
             search_query = search_query.exclude(
                 unlisted_organization__exact='').filter(
                     unlisted_organization__icontains=organization)
-
+    if event:
+        search_query = search_query.filter(shift__job__event__name__icontains=event)
```
Comment (abhishekarya286) [inline vms/volunteer/services.py]: Regarding my comment, we have a problem here. Please resolve this.

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 Fixed it for both event and job fields please review.

**Comment by abhishekarya286** [comment]
@Monal5031 please take a final pass.

**Comment by abhishekarya286** [comment]
Thanks @anjali-dhanuka

**Pull request merged**